go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -357,26 +356,37 @@ func newOvnAddressSet(nbClient libovsdbclient.Client, name string, ips []net.IP)
 				name, err)
 		}
 	} else {
-		//create a new addressSet
-		ops, err := nbClient.Create(&nbdb.AddressSet{
-			Name:        as.hashName,
-			Addresses:   uniqIPs,
-			ExternalIDs: map[string]string{"name": as.name},
+		ops := make([]ovsdb.Operation, 0, 2)
+		ops = append(ops, ovsdb.Operation{
+			Op:      ovsdb.OperationWait,
+			Timeout: 0,
+			Table:   "Address_Set",
+			Where:   []ovsdb.Condition{{Column: "name", Function: ovsdb.ConditionEqual, Value: name}},
+			Columns: []string{"name"},
+			Until:   "!=",
+			Rows:    []ovsdb.Row{{"name": name}},
 		})
+		if addrSet.UUID == "" {
+			// hack used to make TransactAndCheckAndSetUUIDs track the model correctly
+			addrSet.UUID = libovsdbops.BuildNamedUUID()
+		}
+		// create a new addressSet
+		op, err := nbClient.Create(addrSet)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create address set %s (%v)",
 				name, err)
 		}
-		results, err := libovsdbops.TransactAndCheck(nbClient, ops)
+		ops = append(ops, op...)
+		results, err := libovsdbops.TransactAndCheckAndSetUUIDs(nbClient, addrSet, ops)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create new address set %s (%v)",
 				name, err)
 		}
-		if len(results) == 1 {
-			as.uuid = results[0].UUID.GoUUID
-		} else {
-			//should never happen
-			return nil, fmt.Errorf("returned too many results from addressSet creation")
+		as.uuid = addrSet.UUID
+		if len(as.uuid) == 0 {
```

Comment:
nit: for consistency sake, may we use `if as.UUID == ""` like in line 369 ?

Reply:
I think either way is acceptable